baseplate/lib/experiments/providers/simple_experiment.py

```diff
@@ -168,14 +178,25 @@ def __init__(
 
         self.variant_set = variant_set
 
-        self._seed = bucket_seed
-        if self._seed is None:
+        if bucket_seed:
```

Comment:
An example of reordering `if`s a bit to convince the type checker we're handling `None`s properly